frontend/src/sorted_table.tsx

```diff
@@ -0,0 +1,311 @@
+/**
+ * @file This file contains the SortedTable function. This function is a
+ * wrapper around MaterialTable, where you can sort the table by the column
+ * values.
+ */
+
+import React from 'react'
+import { CircularProgress, makeStyles, TableHead, TableRow, TableCell,
+        TableSortLabel, TableContainer, Table, TablePagination, TableBody, Button,
+        } from '@material-ui/core';
+
+const useStyles = makeStyles({
+  table: {
+    padding: "5px",
+    overflowY: "scroll",
+  },
+  columnLabel: {
+    paddingRight: "30px",
+  },
+  columnSelect: {
+    zIndex: 15,
+  },
+  loading: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  functionLink: {
+    color: 'blue',
+    textDecoration: 'underline',
+    cursor: 'pointer',
+  },
+  sorted: {
+    border: 0,
+    clip: 'rect(0 0 0 0)',
+    height: 1,
+    margin: -1,
+    overflow: 'hidden',
+    padding: 0,
+    position: 'absolute',
+    top: 20,
+    width: 1,
+  },
+  bottomButton: {
+    float: 'left',
+  }
+});
+
+/**
+ * Column definition with sorting information. `sortBy` is used to tell the
+ * sorting function which sub-field to sort by if the column value is an object,
+ * and `sortAs` is used to coerce the value to a new type.
+ */
+export interface Column {
+  field: any;
+  label: string;
+  sortBy?: string;
+  sortAs?: 'number' | 'numbers' | 'date' | undefined;
+  hidden?: boolean;
+  numeric?: boolean;
+  render?: Function;
+}
+type Order = 'asc' | 'desc'
+
+/**
+ * Sorts a list of individuals by the `orderBy` field and `order` direction, as
+ * well as the `column` definition information `sortBy` and `sortAs`.
+ * When the value of a field is an object, `sortBy` is used to select which
+ * subfield to use for sorting. `sortAs` can be `number`, `numbers` or `date`
+ * to force a column to be sorted as a particular type.
+ * - `date` fields will be cast as dates using `new Date(<value)`
+ * - `number` fields will be cast as numbers by removing all non-numerical
+ *   characters. ex. `G111-222` -> `111222`
+ * - `numbers` fields are cast as numbers, but using the first set of numbers
+ *   as the integer part, and the remaning as decimals.
+ *   ex. `G111-222-333` -> `111.222333`.
+ *
+ * @param a First object
+ * @param b Second object
+ * @param column Column definition of the table to be sorted
+ * @param orderBy sub-field to order by for object values
+ * @param order `asc` or `desc` for ascending or descending ordering
+ */
+function columnSort(a: any, b: any,  column: Column, orderBy: string, order: Order ) {
```

Comment:
Can we share code from `filter_table.tsx:individualSort`? It seems to be roughly the same

Reply:
as noted in the description - yeah, this code is very similar to FilterTable - The idea is to make FilterTable extend SortedTable in another PR! :)

Reply:
Alright!